src/main/java/com/faforever/client/remote/FafServerAccessorImpl.java

```diff
@@ -427,8 +432,22 @@ private void onSessionInitiated(SessionMessage sessionMessage) {
   }
 
   private void logIn(String username, String password) {
-    String uniqueId = uidService.generate(String.valueOf(sessionId.get()), preferencesService.getFafDataDirectory().resolve("uid.log"));
-    writeToServer(new LoginClientMessage(username, Hashing.sha256().hashString(password, UTF_8).toString(), sessionId.get(), uniqueId, localIp));
+    try {
+      String uniqueId = uidService.generate(String.valueOf(sessionId.get()), preferencesService.getFafDataDirectory().resolve("uid.log"));
+      writeToServer(new LoginClientMessage(username, Hashing.sha256().hashString(password, UTF_8).toString(), sessionId.get(), uniqueId, localIp));
+    } catch (IOException e) {
+      onUIDNotFound(e);
+    }
+  }
+
+  @Override
+  protected void onUIDNotFound(Exception e) {
+    if (e.getMessage() == null) {
+      return;
+    }
```

Comment:
We shouldn't swallow exceptions just because there is no message. At least log it, additionally we can display an error to the user just without the message.